### ISSUE TITLE
[suggestion] Add Command key (Mac) as override key for Automate Quests (instructions included)

### ISSUE THREAD

**Issue by tflo** [leatrix/leatrix-plus#88]
On the Mac, the [Command key](https://en.wikipedia.org/wiki/Command_key) is the main modifier key (and also the [most comfortable one to access](https://3.bp.blogspot.com/-vwhDS5Pz3i0/V9C-X5EB7tI/AAAAAAAANvo/G7vGHVpqe_UzCeKqoK5Cnw_KTrOT4lv0wCK4B/w1200-h630-p-k-no-nu/apple-11q3-wirelesskeyboard-top-lg.jpg)). Would be great to have it added as option for the Automate Quest “override key”.

![WoWScrnShot_080722_161452 scaled](https://user-images.githubusercontent.com/1410282/183295259-7e46d690-dabb-4743-82c5-2aee7554ce05.png)

The Command key has not always been available as modifier key inside WoW. Blizz [introduced this](https://us.forums.blizzard.com/en/wow/t/use-command-as-control-key-missing-in-ptr/652450/2) only with 9.0. Many addons (for example [Dominos](https://github.com/tullamods/Dominos/issues/465), [Clicked](https://github.com/Snakybo/Clicked/issues/84), [LibKeyBound](https://www.curseforge.com/wow/addons/libkeybound-1-0/files/3368631)) have adopted the key in the meantime.

#### To do this, you only have to modify/add three lines in Leatrix_Plus.lua:

1. Modify line 2417:

```lua
LeaPlusLC:CreateDropDown("AutoQuestKeyMenu", "Override key", QuestPanel, 146, "TOPLEFT", 356, -115, {L["SHIFT"], L["ALT"], L["CONTROL"], L["CMD (Mac)"]}, "")
```

2. Add new line 2465:

```lua
or LeaPlusLC["AutoQuestKeyMenu"] == 4 and IsMetaKeyDown()
```

3. Modify line 11410 (= 11411 after step 2):

```lua
LeaPlusLC:LoadVarNum("AutoQuestKeyMenu", 1, 1, 4)
```

***

I haven’t done this in my fork, but you could also test for the WoW client platform, and suppress the CMD drop-down menu entry for non-Mac clients (and remove the “(Mac)” part). You can test with this function:

```lua
IsMacClient()
```

*** 

A note on the name of the key:

The WoW API uses the Linux term “Meta”. In the WoW client GUI (Key Bindings) they use “CMD”. In macro conditionals we can use `mod:cmd` as well as `mod:meta`. The official name of the modifier is “Command” or short “Cmd”.

Thus I have proposed “CMD” for the drop-down menu. (“Meta” is not used at all in the Mac world.)

***

 Thanks in advance for considering this little improvement

– Tom

**Comment by leatrix** [comment]
Hey Tom,
Thanks for the very thorough post.  Added to the latest alpha releases.

**Comment by tflo** [comment]
Thank you for implementing it!

I have pointed you to the `IsMacClient()` API function without ever having tested it (bad idea to blindly trust a function created by Blizz ;)

But it seems to work, at least on the Mac side. For example:

```lua
local autoQuestKeys = {L["SHIFT"], L["ALT"], L["CONTROL"]}
if IsMacClient() then table.insert(autoQuestKeys, L["CMD"]) end
LeaPlusLC:CreateDropDown("AutoQuestKeyMenu", "Override key", QuestPanel, 146, "TOPLEFT", 356, -115, autoQuestKeys, "")

```
 
I cannot test it for Windows, but I tried the “opposite” function `IsWindowsClient()` and it works as expected on a Mac (i.e. it returns false).
